pkg/apiserver/resthandler.go

```diff
@@ -308,13 +305,9 @@ func (h *RESTHandler) handleRESTStorage(parts []string, req *http.Request, w htt
 }
 
 // createOperation creates an operation to process a channel response.
-func (h *RESTHandler) createOperation(out <-chan RESTResult, sync bool, timeout time.Duration, onReceive func(RESTResult)) *Operation {
+func (h *RESTHandler) createOperation(out <-chan RESTResult, timeout time.Duration, onReceive func(RESTResult)) *Operation {
 	op := h.ops.NewOperation(out, onReceive)
-	if sync {
-		op.WaitFor(timeout)
-	} else if h.asyncOpWait != 0 {
-		op.WaitFor(h.asyncOpWait)
-	}
+	op.WaitFor(timeout)
```

Comment:
Wouldn't it just be easier to remove operation completely, and replace it with the code from op.WaitFor with a simple timeout?  We already have the http level max client timeout on the server, and I'd like to remove the extra locking operations on Operations and the two go routines it creates.